moztelemetry/spark.py

```diff
@@ -145,6 +150,92 @@ def get_one_ping_per_client(pings):
                     reduceByKey(lambda p1, p2: p1).\
                     map(lambda p: p[1])
 
+def get_data_sources():
+    try:
+        key = _bucket_meta.get_key("sources.json")
+        sources = key.get_contents_as_string()
+        return json.loads(sources)
+    except ssl.SSLError:
+        return {}
+
+def get_source_schema(source_name):
+    global _sources
+    if _sources is None:
+        _sources = get_data_sources()
+
+    if source_name not in _sources:
+        raise ValueError("Unknown data source: {}. Known sources: [{}].".format(source_name, ", ".join(sorted(_sources.keys()))))
+
+    if "schema" not in _sources[source_name]:
+        try:
+            key = _bucket_meta.get_key("{}/schema.json".format(_sources[source_name].get("metadata_prefix", source_name)))
+            schema = key.get_contents_as_string()
+            _sources[source_name]["schema"] = json.loads(schema)
+        except ssl.SSLError:
+            return None
+    return _sources[source_name]["schema"]
+
+def get_records(sc, source_name, **kwargs):
```

Comment:
This public function should be located at the top of the files, with the other public functions.